src/snowflake/snowpark/session.py

```diff
@@ -2241,22 +2323,37 @@ def write_pandas(
                         "snowflake-connector-python to 3.4.0 or above.",
                         stacklevel=1,
                     )
-            success, nchunks, nrows, ci_output = write_pandas(
-                self._conn._conn,
-                df,
-                table_name,
-                database=database,
-                schema=schema,
-                chunk_size=chunk_size,
-                compression=compression,
-                on_error=on_error,
-                parallel=parallel,
-                quote_identifiers=quote_identifiers,
-                auto_create_table=auto_create_table,
-                overwrite=overwrite,
-                table_type=table_type,
-                **kwargs,
-            )
+            if "modin" in str(type(df)):
```

Comment:
I don't think the intention of this line is very clear. Something like this would be more clear:
```suggestion
            if isinstance(df, (snowflake.snowpark.modin.pandas.DataFrame, snowflake.snowpark.modin.pandas.Series):
```

Reply:
This won't work because modin is not always imported.

Reply:
rewrite this part now PTAL